var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -332,6 +333,9 @@ class Mpich(MpichEnvironmentModifications, AutotoolsPackage, CudaPackage, ROCmPa
 
     depends_on("pmix", when="pmi=pmix")
 
+    # Libevent is required when *vendored* PMIx is used
```

Comment:
We don't vendor openpmix in MPICH, so this dependency might be coming from an external PMIx library?

Reply:
@raffenet I got the solution through this PR https://github.com/spack/spack/pull/41686. From my understanding, the vendored term here meant an external vendor PMIx (such as cray). But maybe I misunderstood. Happy to change it to external if you think that's more appropriate.

Reply:
Got it. So the dependency is from the external library. I'll suggest a change in the comment because "vendoring" to me means the opposite i.e. an internal copy is used rather than an external one.